app/internal/pay/client_test.go

```diff
@@ -295,6 +294,23 @@ func TestGetPayment(t *testing.T) {
 		assert.Equal(t, GetPaymentResponse{}, actualGPResponse, "Return value did not match")
 	})
 
+	t.Run("Returns an error if unable to make request", func(t *testing.T) {
+		server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {}))
+
+		defer server.Close()
+
+		payClient := Client{BaseURL: "not an url", ApiKey: apiToken, HttpClient: server.Client()}
+
+		actualGPResponse, err := payClient.GetPayment(paymentId)
+
+		if err == nil {
```

Comment:
you can use `if assert.NotNil(t, err) { ... }` when you want to guard other assertions behind another

Reply:
or use `assert.ErrorContains(t, err, "the error message")`